### PR TITLE
add unsupervised learning

### PR DESCRIPTION
**Pull request by Zhaochengxiann**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [ ] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb [hunk 1]
```diff
@@ -0,0 +1,674 @@
+{
```
Comment (reveurmichael) [inline open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:1]: <p>move line 6 to line 1</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/195/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (reveurmichael) [inline open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:1]: <p>Enmpty line 1, remove</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/195/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='64'/>

Comment (reveurmichael) [reply open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:1]: <p>remove sklearn</p>

Comment (reveurmichael) [inline open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:1]: <p>use pandas.readcsv()</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/195/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (reveurmichael) [inline open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:1]: <p>we don't load the dataset twice.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-academy/machine-learning/pull/195/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

File: open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb [hunk 2]
```diff
@@ -0,0 +1,674 @@
...
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.8.5"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
+}
```
Comment (reveurmichael) [inline open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:674]: Please add

`- file: assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering`

to 

`_toc.yml`

File: open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb (hunk 1 above)
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:1]: <p>thanks</p>

[the remark this answers is not in the document]
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:1]: <p>i'll format it， thanks</p>

[the remark this answers is not in the document]
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:1]: <p>ok, done</p>

Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:1]: ok

File: open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb (hunk 2 above)
Comment (Zhaochengxiann) [reply open-machine-learning-jupyter-book/assignments/ml-advanced/unsupervised-learning/unsupervised-intro-with-kmeans-clustering.ipynb:674]: done

**Comment by Zhaochengxiann** [comment]
#217 #236

**Review by reveurmichael — APPROVED** [review]

**Pull request merged**